cpp/open3d/data/Extract.h

```diff
@@ -0,0 +1,41 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018-2021 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <string>
+
+namespace open3d {
+namespace data {
+
+/// \brief Function to extract compressed files.
+/// \param filename Filename including path-prefix.
+/// Example: "/path/to/file/file.zip"
```

Comment:
Is this a file name or file path?
Does the `extract_dir` contain the prefix?
Shall we use the absolute path directly, since this is not used by the user?

Reply:
Yes, `extract_dir` (file path) is the absolute path, it contains `data_root/prefix`. 
`filename` contains `path/to/file/filename.zip` (so, its `filename_with_prefix`). 
Do you have some suggestions to make it more clear?

Reply:
### Comments

- `name` can be confusing, since `name` typically means something different from `path`.
- Whether it is a relative path or an absolute path, we shall keep both arguments consistent.

### Suggestions

- Option 1: 
  - `const std::string& file_path`, `const std::string& extract_dir`
  - Both use absolute path. That is do not add any prefix in the `Extract()` function.
- Option 2:
  - `const std::string& file_path_from_data_root`, `const std::string& extract_dir_from_data_root`.
  - It is fine that the name is long since the user doesn't see it.

I prefer Option 1. Since it makes the `Extract()` function decouples from the dataset class design.

Reply:
Done. Thanks for the detailed suggestion.